nixos/modules/services/networking/sing-box.nix

```diff
@@ -88,6 +88,8 @@ in
           "${lib.getExe cfg.package} -D \${STATE_DIRECTORY} -C \${RUNTIME_DIRECTORY} run"
         ];
       };
+      # After= is specified by upstream
+      wants = [ "network-online.target" ];
```

Comment:
There should be an assertion which forces you to also set requires network-online.target